dispatch/dispatch.go

```diff
@@ -136,7 +140,11 @@ func (d *Dispatcher) Run() {
 	d.aggrGroupsPerRoute = map[*Route]map[model.Fingerprint]*aggrGroup{}
 	d.aggrGroupsNum = 0
 	d.metrics.aggrGroups.Set(0)
-	d.ctx, d.cancel = context.WithCancel(context.Background())
+	ctx := context.Background()
+	if d.alertLCObserver != nil {
+		ctx = notify.WithAlertLCObserver(ctx, d.alertLCObserver)
```

Comment:
should we put the observer into the stages rather than in ctx?

Reply:
You mean pass it as one of the arguments in the Exec call instead of adding it in the context?